service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt

```diff
@@ -355,86 +361,107 @@ SELECT
     ) AS upcoming_placement_type
 FROM person p
 LEFT JOIN child_images ci ON ci.child_id = p.id
-WHERE p.id = ANY (:childIds)
+WHERE p.id = ANY (SELECT child_id FROM children)
 ORDER BY p.date_of_birth, p.duplicate_of
         """
         )
-        .bind("childIds", childIds)
+        .bind("guardianId", guardianId)
         .bind("today", today)
         .mapTo<ReservationChild>()
         .list()
 }
 
 data class ReservationPlacement(
-    val isBackup: Boolean,
     val childId: ChildId,
-    val startDate: LocalDate,
-    val endDate: LocalDate,
+    val range: FiniteDateRange,
+    val type: PlacementType,
     val operationDays: Set<Int>,
-    val hasContractDays: Boolean
-)
-
-data class ReservationPlacements(
-    val backupPlacements: List<ReservationPlacement>,
-    val regularPlacements: List<ReservationPlacement>
 )
 
 fun Database.Read.getReservationPlacements(
-    guardianId: PersonId,
-    today: LocalDate,
+    childIds: Set<ChildId>,
```

Comment:
nitpick: getPlacementsWithReservationsEnabled could make more sense

Reply:
maybe this could then already also filter by placement type?

Reply:
All placements (with reservations enabled in the unit) are needed regardless of whether the placement type requires reservations or not. Otherwise the child wouldn't be visible in the calendar to mark absences.